File: app/src/main/java/com/left/gank/ui/discovered/jiandan/JiandanContract.java
package com.left.gank.ui.discovered.jiandan;

import android.content.Context;

import androidx.annotation.NonNull;

import com.left.gank.domain.JianDanBean;
import com.left.gank.mvp.base.LoadMorePresenter;
import com.left.gank.mvp.base.SupportView;

import java.util.List;

/**
 * Create by LingYan on 2016-11-21
 */

public interface JiandanContract {
    interface View extends SupportView {
        void refillData(List<JianDanBean> list);

        void appendMoreDate(List<JianDanBean> list);
    }

    abstract class Presenter extends LoadMorePresenter<View> {

        public Presenter(@NonNull Context context, View view) {
            super(context, view);
        }
    }
}

File: app/src/main/java/com/left/gank/ui/discovered/jiandan/JiandanContract.kt
package com.left.gank.ui.discovered.jiandan

import android.content.Context

import com.left.gank.domain.JianDanBean
import com.left.gank.mvp.base.LoadMorePresenter
import com.left.gank.mvp.base.SupportView

/**
 * Create by LingYan on 2016-11-21
 */

interface JiandanContract {
    interface View : SupportView {
        fun loadJiandanSuccess(page: Int, list: List<JianDanBean>?)

        fun loadJiandanFailure(page: Int, msg: String)
    }

    abstract class Presenter(context: Context, view: View) : LoadMorePresenter<View>(context, view) {
        abstract fun loadJianDan(page: Int)
    }
}
